File: fw/wrappers/javascript/sample.js
/* INFO **
** INFO */

(function ()
{
    'use strict';

    print('>>> Starting...');

    /* Create "actors" */
    var context  = new kb.rpi2.Context(),
        event    = new kb.rpi2.Event(context),
        led      = new kb.rpi2.LED(event, kb.rpi2.PIN1),
        fourKeys = new kb.rpi2.FourKeys(event, {pinRow  : kb.rpi2.PIN2,
                                                pinCol1 : kb.rpi2.PIN3,
                                                pinCol2 : kb.rpi2.PIN4,
                                                pinCol3 : kb.rpi2.PIN5,
                                                pinCol4 : kb.rpi2.PIN6});

    /* Bind callbacks */
    fourKeys.onKey3 = function (fourKeys, event, context)
    {
        print('fourKeys.onKey3()');
    };
    fourKeys.onEnable = function (fourKeys, event, context)
    {
        print('fourKeys.onEnable()');
    };

    led.onOn = function (led, event, context)
    {
        print('led.onOn()');
    };

    context.onCycleBegin = function (context, event)
    {
        print('context.onCycleBegin()');
        context.onCycleEnd = function (context, event)
        {
            print('context.onCycleEnd()');
            context.stop();
        };
    };
    context.onStop = function (context, event)
    {
        print('context.onStop()');
    };

    /* Enter event loop */
    event.activate();
    context.start();

    print('<<< Exiting...');
})();

File: fw/wrappers/python/sample.py
## INFO ##
## INFO ##

from kb.rpi2 import (Context,
                     Event,
                     LED,
                     FourKeys,
                     PIN1,
                     PIN2,
                     PIN3,
                     PIN4,
                     PIN5,
                     PIN6)

print('>>> Starting...')

context   = Context()
event     = Event(context)
led       = LED(event, PIN1)
four_keys = FourKeys(event, pin_row  = PIN2,
                            pin_col1 = PIN3,
                            pin_col2 = PIN4,
                            pin_col3 = PIN5,
                            pin_col4 = PIN6)

@event
def on_key_3(for_keys, event, context):
    print('four_keys.on_key_3()')

@event
def on_enable(four_keys, event, context):
    print('four_keys.on_enable()')

@led
def on_on(led, event, context):
    print('led.on_on()')

@context
def on_cycle_begin(context, event):
    print('context.on_cycle_begin()')
    @context
    def on_cycle_end(context, event):
        print('context.on_cycle_end()')
        context.stop()

@context
def on_stop(context, event):
    print('context.on_stop()')

event.activate()
context.start()

print('<<< Exiting...')
